app/src/main/java/fr/free/nrw/commons/category/CategoryImagesListFragment.java

```diff
@@ -269,6 +270,37 @@ private void handleSuccess(List<Media> collection) {
         progressBar.setVisibility(GONE);
         isLoading = false;
         statusTextView.setVisibility(GONE);
+        for (Media m : collection) {
+            replaceTitlesWithCaptions(m.getDisplayTitle(), mediaSize++);
+        }
+    }
+
+    /**
+     * fetch captions for the image using filename and replace title of on the image thumbnail(if captions are available)
+     * else show filename
+     */
+
+    public void replaceTitlesWithCaptions(String displayTitle, int i) {
+        compositeDisposable.add(mediaClient.getCaptionByFilename("File:" + displayTitle+ ".jpg")
```

Comment:
is it always `.jpg`? Aren't some files are `.jpeg` or `.PNG` for instance?

Reply:
I tested on many and I found only "jpg" hence I added this extension.
Are there files with the other extensions?

Reply:
I am not sure, if this will work for all files, are all file name for sure ".png"?

Reply:
(see my comment 10 lines above)

Reply:
(after solving this, please apply same resolution at DepictedImagesPresenter.java)